Assets/Scripts/Movement/MousePositionAndLook.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEditor.Animations;
+using UnityEngine;
+
+public class MousePositionAndLook : MonoBehaviour
+{
+    // speed is the rate at which the object will rotate
+    public float speed;
+
+    private void Start()
+    {
+        Cursor.lockState = CursorLockMode.Confined;
+    }
+
+    void FixedUpdate()
+    {
+        Vector3 mousePos = Input.mousePosition;
+        Debug.Log($"mousePosition: {mousePos}");
+
+        Vector3 worldPoint = Camera.main.ScreenToWorldPoint(new Vector3(mousePos.x, mousePos.y, Camera.main.nearClipPlane));
+        Debug.Log($"point: {worldPoint}");
```

Comment:
You should look at ScreenToWorldPoint's overloads. I'm assuming your issue is caused by you not providing a proper depth value for it to trace the world position to. I would cast a single ray everyframe in the direction of camera.forward starting from the position of the camera.

Reply:
float dist;
RaycastHit hit;
Ray forward = new Ray(Camera.main.transform.position, Camera.main.transform.forward);
        
if(Physics.Raycast(forward, out hit))
{
    dist = hit.distance;
}

Reply:
You can use the "dist" float instead of the camera's near clip plane in your world point function. That should return the proper world position of your mouse cursor. Note that if a wall gets in the way of the camera this might become an issue. You could assign a physics layer to walls that the raycast would ignore to circumvent this issue.